lib/ansible/modules/cloud/vmware/vmware_dvswitch_nioc.py

```diff
@@ -0,0 +1,189 @@
+author:
+    - Joseph Andreatta (@vmwjoseph) <joseph () vmware.com>
+
+notes:
+    - Tested on vSphere 6.7
+requirements:
+    - "python >= 2.6"
+    - PyVmomi
+options:
+    switch_name:
+        description:
+            - The name of the switch to create or remove
+        required: True
+    version:
+        description:
+            - Network IO control version
+        default: 'version2'
+        choices:
+            - 'version2'
+            - 'version3'
+        required: False
+    state:
+        description:
+            - Create or remove dvSwitch
+        default: 'present'
+        choices:
+            - 'present'
+            - 'absent'
+        required: False
+extends_documentation_fragment: vmware.documentation
+'''
+
+RETURN = ''' # '''
+
+EXAMPLES = '''
+- name: Enable NIOC
+  local_action:
+    module: vmware_dvswitch_nioc
+    hostname: vcenter_ip_or_hostname
+    username: vcenter_username
+    password: vcenter_password
+    switch_name: dvSwitch
+    version: version3
+    state: present
+'''
+
+try:
+    from pyVmomi import vim, vmodl
+    HAS_PYVMOMI = True
+except ImportError:
+    HAS_PYVMOMI = False
+
+from ansible.module_utils.basic import AnsibleModule
+from ansible.module_utils.vmware import (HAS_PYVMOMI,
+                                         connect_to_api,
+                                         find_datacenter_by_name,
+                                         find_dvs_by_name,
+                                         vmware_argument_spec,
+                                         wait_for_task
+                                         )
+
+
+class VMwareDVSwitchNIOC(object):
+
+    def __init__(self, module):
+        self.module = module
+        self.dvs = None
+        self.switch_name = self.module.params['switch_name']
+        self.version = self.module.params['version']
+        self.state = self.module.params['state']
+        self.content = connect_to_api(module)
+
+    def process_state(self):
+        dvs_states = {
+            'absent': {
+                'present': self.state_disable_nioc,
+                'absent': self.state_exit_unchanged,
+            },
+            'present': {
+                'update': self.state_update_nioc,
+                'present': self.state_exit_unchanged,
+                'absent': self.state_enable_nioc,
+            }
+        }
+        dvs_states[self.state][self.check_nioc_state()]()
+
+    def state_exit_unchanged(self):
+        self.module.exit_json(changed=False)
+
+    def state_disable_nioc(self):
+        changed = True
+        result = "Disable NIOC"
+
+        if not self.module.check_mode:
+            task = self.dvs.EnableNetworkResourceManagement(enable=False)
+
+        self.module.exit_json(changed=changed, result=str(result))
+
+    def state_enable_nioc(self):
+        changed = True
+        result = "Enable NIOC"
+
+        if not self.module.check_mode:
+            task = self.dvs.EnableNetworkResourceManagement(enable=True)
```

Comment:
If you provide non-existing DVSwitch then `self.dvs` will be `None` and module will fail with error saying 

```
"'NoneType' object has no attribute 'EnableNetworkResourceManagement'"
```